lib/puppet/util/windows/api_types.rb

```diff
@@ -0,0 +1,278 @@
+require 'ffi'
+require 'puppet/util/windows/string'
+
+module Puppet::Util::Windows::APITypes
```

Comment:
Since the PDK needs to be able to deal with multiple versions of puppet, the main binary will never reference the puppet gem.

For the record: I would prefer that utility code to live in a independent gem that others can use without having to depend on a full puppet package. But this is not relevant to this PR. In effect, we're treating this like a vendoring.